### PR TITLE
Fixed error on View Order when contains a customized product which has been deleted

### LINKED ISSUE #25644: BO - Order page - Debug mode enabled - an exception is displayed when ordering a deleted product with a file customization
#### Describe the bug

In the BO > Orders > Order details page, an exception is displayed when ordering a deleted product with a file customization

#### Steps to Reproduce

1. Create a product with a customization
- Type: file
- Label: valid label
- Required (yes/No) => Same issue
2. Go to FO > Product details page
3. Choose a file and save a customization
4. add to cart
5. Continue checkout & Place Order
6. Go to BO > Catalog > Products page
7. Delete the product created
8. Go o BO > Orders > View last order created
9. See error 

image

In the PS1769 => ok

#### Additional information

* PrestaShop version: 1775, 1776, 1777build1
* PHP version: 7.1, 7.2

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Fixed error on View Order when contains a customized product which has been deleted
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25644
| How to test?      | Cf #25644

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25704)

### CODE REVIEW

**Review by matthieu-rolland — APPROVED** [review]

**Review by sowbiba — COMMENTED** [review]
Shouldnt we delete the customization that is no longer available ?

**Comment by Progi1984** [comment]
@sowbiba The customization of the product is removed. But not the customized customization of the ordered product.

File: tests/Integration/Behaviour/Features/Scenario/Order/order_product.feature [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      | productReference | type | name       | value |
+      | product12345     | text | back image | Toto  |
+
+    ## Remove product
+    When I delete product product12345
+    Then product product12345 should not exist anymore
+    ## Check customization
+    Then the order "bo_order1" should have following customizations:
+      | productReference | type | name       | value |
+      | product12345     | text |            | Toto  |
```
Comment (atomiix) [inline tests/Integration/Behaviour/Features/Scenario/Order/order_product.feature:59]: I know it's probably out of scope but isn't that weird that we don't keep the name of the customization?

Comment (Progi1984) [reply tests/Integration/Behaviour/Features/Scenario/Order/order_product.feature:59]: It's out of scope. But I can explain : 

The customization of the product is removed (and its label). But not the customized customization of the ordered product.

**Review by atomiix — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM thanks @Progi1984  !!

**Comment by Progi1984** [comment]
Thanks @SD1982

**Pull request merged**